numpy/core/numeric.py

```diff
@@ -1226,11 +1226,12 @@ def rollaxis(a, axis, start=0):
     a : ndarray
         Input array.
     axis : int
-        The axis to roll backwards.  The positions of the other axes do not
+        The axis to roll backwards. The positions of the other axes do not
         change relative to one another.
     start : int, optional
-        The axis is rolled until it lies before this position.  The default,
-        0, results in a "complete" roll.
+        The axis is moved until it lies before this position. The default
+        value is 0. If `axis == start`, the input array remains the same. 
```

Comment:
You need to be careful here that these conditions only apply to _positive_ `start` and `axis`. We also need a description of the rules for converting `start` from negative to positive, something which did not follow the usual rules.

Reply:
In particular, the particulary nasty wart of the function is:

| in | out |
|--|--|
| `-(arr.ndim+1)` | AxisError |
| `-arr.ndim` | `0` |
| ... | ... |
| `-1` | `arr.ndim-1` |
| `0` | `0` |
| ... | ... |
| `arr.ndim` | `arr.ndim` |
| `arr.ndim + 1` | AxisError  |


Note how it is impossible to specify `start=arr.ndim` via a negative index.

Reply:
Thanks for laying out the specifics in that table. I know deprecation was rejected in #9475, but I would keep that table in mind for the future if/when the conversation about deprecating `rollaxis` is ever picked back up.

In the meantime, since deprecation is out, I would vote that the above table should be included in the docstring.

Reply:
What would be the correct way to put this table in the docs? Should it go inside the **Parameters** section?

Reply:
Good question - my main motivation for including the table is to convince people to use `moveaxis` instead of `rollaxis` :). I suppose it makes sense in the Parameters section, even though it's so long. Maybe others have different/better opinions.

Reply:
Put the table right after `start` and this is how it is being rendered. 

![image](https://user-images.githubusercontent.com/2712115/79804882-4e687880-833b-11ea-94ca-6a7cb5c0da31.png)